tests/test_TFNetworkLayer.py

```diff
@@ -981,33 +981,53 @@ def test_SplitDimsLayer_resolve_dims():
   assert_equal(SplitDimsLayer._resolve_dims(old_dim=2 * 3 * 5, new_dims=(2, 3, -1, 1)), (2, 3, 5, 1))
 
 
-def _check_MergeDimsLayer(session, in_data_opts, in_static_shape, opts, out_data_shape, out_static_shape):
+def _check_MergeDimsLayer(session, in_data_opts, in_static_shape, opts, out_data_shape, out_static_shape,
+                          out_size_placeholder=None):
   """
   :param TFCompat.v1.Session session:
   :param dict[str] in_data_opts:
   :param tuple[int] in_static_shape:
   :param dict[str] opts: for MergeDimsLayer
   :param tuple[int|None] out_data_shape:
   :param tuple[int] out_static_shape:
+  :param dict[int, tuple[int]] out_size_placeholder:
   :rtype: MergeDimsLayer
   """
   net = TFNetwork(extern_data=ExternData())
   rnd = numpy.random.RandomState(42)
   src = InternalLayer(name="src", network=net, out_type=in_data_opts)
   print("input:", src.output)
   src.output.placeholder = tf.constant(rnd.normal(size=in_static_shape).astype("float32"), dtype=tf.float32)
-  src.output.size_placeholder = {}  # not sure if enough...
+  src.output.size_placeholder = {}
+
+  if out_size_placeholder:
+    n_batch = in_static_shape[src.output.batch_dim_axis]
+    for axis, dim in enumerate(in_data_opts['shape']):
+      if dim is None:
+        n_time = in_static_shape[src.output.get_batch_axis(axis)]
+        dyn_size_v = numpy.array([n_time, max(n_time - 2, 1), max(n_time - 3, 1)])
+        if dyn_size_v.shape[0] > n_batch:
+          dyn_size_v = dyn_size_v[:n_batch]
+        elif dyn_size_v.shape[0] < n_batch:
+          assert False, "batch_size > 3 not available for testing size_placeholder"
+        src.output.size_placeholder[axis] = tf.constant(dyn_size_v)
```

Comment:
Why not simply:
```suggestion
        src.output.size_placeholder[axis_wo_b] = tf.fill([n_batch], in_static_shape[axis])
```
Keep it simple.

Reply:
I copied this from `make_feed_dict`, and I thought the test might be better if there are sequences of different length.